pkg/helper/client.go

```diff
@@ -214,33 +214,37 @@ func checkApiServerConnectivity(timeout time.Duration) error {
 		config.Timeout = timeout
 		client, err := client.New(config, client.Options{})
 		if err != nil {
-			return false, errors.Wrap(err, "creating new custom client")
+			log.Error(err, "failed to creating new custom client")
+			return false, nil
 		}
 		err = client.Get(context.TODO(), types.NamespacedName{Name: metav1.NamespaceDefault}, &corev1.Namespace{})
 		if err != nil {
-			return false, errors.Wrap(err, "reaching to apiserver failed")
+			log.Error(err, "failed reaching the apiserver")
+			return false, nil
 		}
 		return true, nil
 	})
 }
 
 func defaultGw() (string, error) {
 	defaultGw := ""
-	return defaultGw, wait.PollImmediate(1*time.Second, defaultGwRetrieveTimeout, func() (bool, error) {
+	return defaultGw, wait.PollImmediate(1*time.Second, defaultGwRetrieveTimeout*time.Second, func() (bool, error) {
 		observedStateRaw, err := show()
 		if err != nil {
-			return false, fmt.Errorf("error running nmstatectl show: %v", err)
+			log.Error(err, fmt.Sprintf("failed retrieving current state"))
+			return false, nil
 		}
 
 		currentState, err := yaml.YAMLToJSON([]byte(observedStateRaw))
 		if err != nil {
-			return false, fmt.Errorf("Impossible to convert current state to JSON: %v", err)
+			return false, errors.Wrap(err, "failed to convert current state to JSON")
```

Comment:
So in this case we want to cancel the poll?

Reply:
I suppose not being able to transform yaml to json is not transitory, but don't know what will happy with empty, since maybe empty can be transitory value.